src/components/RTsort/RTsort.tsx

```diff
@@ -17,28 +18,64 @@ interface FormValuesType {
 const rtsort = () => {
     const [loading, setLoading] = useState(false);
     const [output, setOutput] = useState("");
+    const [pid, setPid] = useState("");
 
     let form = useForm({
         initialValues: {
             path: "./",
         },
     });
 
+    // Uses the callback function of runCommandGetPidAndOutput to handle and save data
+    // generated by the executing process into the output state variable.
+    const handleProcessData = useCallback((data: string) => {
+        setOutput((prevOutput) => prevOutput + "\n" + data); // Update output
+    }, []);
+
+    // Uses the onTermination callback function of runCommandGetPidAndOutput to handle
+    // the termination of that process, resetting state variables, handling the output data,
+    // and informing the user.
+    const handleProcessTermination = useCallback(
+        ({ code, signal }: { code: number; signal: number }) => {
+            if (code === 0) {
+                handleProcessData("\nProcess completed successfully.");
+            } else if (signal === 15) {
+                handleProcessData("\nProcess was manually terminated.");
+            } else {
+                handleProcessData(`\nProcess terminated with exit code: ${code} and signal code: ${signal}`);
+            }
+
+            // Clear the child process pid reference
+            setPid("");
+
+            // Cancel the Loading Overlay
+            setLoading(false);
+        },
+        [handleProcessData]
+    );
+
```

Comment:
Need a docstring to describe the purpose of onSubmit. 
```suggestion
/**
     * onSubmit: Asynchronous handler for the form submission event.
     * It sets up and triggers the rt-sort tool with the given parameters.
     * Once the command is executed, the results or errors are displayed in the output.
     *
     * @param {FormValuesType} values - The form value, containing path.
     */
```

- [ ] Add docstring to describe the purpose of onSubmit

Reply:
done